src/docs/stan-reference/examples.tex

```diff
@@ -8046,6 +8046,154 @@ \section{Circular Representations of Days and Years}
+%
+As an illustrative example, we consider a nonlinear system of two equations
+with two unknowns. Our goal is to simultaneously solve all equations for
+$x_1$ and $x_2$, such that the vector function $z$ goes to 0. 
+%
+\begin{eqnarray}\label{algebra.equation}
+  \begin{aligned}
+  z_1 &= x_1 - \theta_1 \\
+  z_2 &= x_1 x_2 + \theta_2 \\
+  \end{aligned}
+\end{eqnarray}
+%
+
+\section{Coding an Algebraic System}
+
+A system of algebraic equations is coded directly in Stan as a function with a
+strictly specified signature. For example, the nonlinear system given by
+\refequation{algebra.equation} can be coded using the following function
+in Stan (see \refchapter{functions-programming} for
+more information on coding user-defined functions).
+%
+\begin{stancode}
+real[] system(real[] x,       // unknowns
+               real[] theta,  // parameters
+               real[] dat_r,    // data (real)
+               real[] dat_i) {  // data (integer)
+  real z[2];
+  z[1] = x[1] - theta[1];
+  z[2] = x[1] * x[2] - theta[2];
+  return z;
+}
+\end{stancode}
+%
+The function takes the unknowns we wish to solve for in \code{x} (a real array),
+the system parameters in \code{theta} (a real array), the real data in
+\code{dat\_r} (a real array) and the integer data in \code{dat\_i} (an integer array). The
+system function returns the value of the function, for which we want to compute the roots. 
+Our example does not use real or integer data. Nevertheless, these
+unused arguments must be included in the system function with exactly the
+signature above.
+
+\subsubsection{Strict Signature}  
+%
+The function defining the system must have exactly these argument types and
+return type. This may require passing in zero-length arrays for data or parameter
+if the system does not involve data or parameters.
+
+\section{Calling the Algebraic Solver}
+%
+Let's suppose $\theta = \{3, 6\}$. To call the algebraic solver, we need to
+provide an initial guess. A good guess can significantly speed up the solver and
+even determine whether the solver converges or not.
+%
+The following code returns the solution to our nonlinear algebraic system:
+%
+\begin{stancode}
+transformed data {
+  real guess[2] = {1, 1};
+  real dat_r[0];
+  int dar_i[0];
+}
+
+transformed parameters {
+  real theta[2] = {3, 6};
+  real x[2];
+
+  x = algebra_solver(guess, theta, dat_r, dat_i);
+}
+\end{stancode}
+
+which returns $x = \{3, -2\}$.
+
+\subsection{Data versus Parameters}
+The arguments for the initial guess \code{guess}, the real data \code{dat\_r}, and
+the integer data \code{dat\_i} must be expressions that only involve data or
+transformed data variables. The parameters \code{theta} is the only argument
+which may involve parameters.
+
+\subsection{Length of the Algebraic Function and of the Vector of Unknowns}
+The Jacobian of the solution with respect to the parameters is computed
+using the implicit function theorem, which places certain restrictions. In particular,
```

Comment:
I think you want "imposes" rather than "places".  With "places", you really need to add a prepositional phrase, such as "places restrictions on Jacobian invertibility".